packages/next-on-pages/src/buildApplication/processVercelFunctions/dedupeEdgeFunctions.ts

```diff
@@ -121,10 +121,10 @@ async function processFunctionIdentifiers(
 				if (newFilePath) identifierPathsToBuild.add(newFilePath);
 				if (newImport) importsToPrepend.push(newImport);
 				if (wasmImports.length) {
-					wasmImportsToPrepend.set(
-						identifierInfo.newDest as string,
-						wasmImports,
-					);
+					const newDest = identifierInfo.newDest as string;
+					const existingWasmImports = wasmImportsToPrepend.get(newDest) ?? [];
+					const newWasmImports = existingWasmImports.concat(wasmImports);
+					wasmImportsToPrepend.set(newDest, newWasmImports);
```

Comment:
Please can you change the `wasmImportsToPrepend` map to use sets instead of arrays so that we ensure there are no duplicates added?

Reply:
😮 yeah! good point! thanks!

Reply:
done 🙂👍